xarray/core/common.py

```diff
@@ -1771,31 +1772,37 @@ def is_np_timedelta_like(dtype: DTypeLike) -> bool:
     return np.issubdtype(dtype, np.timedelta64)
 
 
-def _contains_cftime_datetimes(array) -> bool:
+def _contains_cftime_datetimes(array: Any) -> bool:
     """Check if an array contains cftime.datetime objects"""
-    if cftime is None:
-        return False
+    from xarray.core.variable import Variable
+
+    if isinstance(array, Variable):
+        var = array
     else:
-        if array.dtype == np.dtype("O") and array.size > 0:
-            sample = np.asarray(array).flat[0]
-            if is_duck_dask_array(sample):
-                sample = sample.compute()
-                if isinstance(sample, np.ndarray):
-                    sample = sample.item()
-            return isinstance(sample, cftime.datetime)
-        else:
-            return False
+        var = Variable(dims=tuple(f"dim_{v}" for v in range(array.ndim)), data=array)
+
+    return _variable_contains_cftime_datetimes(var)
 
 
-def contains_cftime_datetimes(var) -> bool:
+def _variable_contains_cftime_datetimes(var: T_Variable) -> bool:
     """Check if an xarray.Variable contains cftime.datetime objects"""
-    if var.dtype == np.dtype("O") and var.size > 0:
-        return _contains_cftime_datetimes(var.data)
-    else:
+    if cftime is None:
         return False
 
+    if var.dtype == np.dtype("O") and var.size > 0:
+        first_idx = tuple(0 for _ in range(var.ndim))
```

Comment:
```suggestion
        first_idx = (0,) * var.ndim
```
to shave off some more ns ;-)